code/parse/sexp.cpp

```diff
@@ -4365,30 +4365,30 @@ int rand_sexp(int n, bool multiple)
 // need to keep.
 int sexp_or(int n)
 {
-    int all_false, result;
+	bool all_false = true;
+	bool result = false;
 
-	all_false = 1;
-	result = 0;
 	if (n != -1)
 	{
 		if (CAR(n) != -1)
 		{
 			result |= is_sexp_true(CAR(n));
-			if ( Sexp_nodes[CAR(n)].value == SEXP_KNOWN_TRUE )
+			if (Sexp_nodes[CAR(n)].value == SEXP_KNOWN_TRUE)
 				return SEXP_KNOWN_TRUE;								// if one of the OR clauses is TRUE, whole clause is true
-			if ( Sexp_nodes[CAR(n)].value != SEXP_KNOWN_FALSE )		// if the value is still unknown, they all can't be false
-				all_false = 0;
+			if (Sexp_nodes[CAR(n)].value != SEXP_KNOWN_FALSE)		// if the value is still unknown, they all can't be false
+				all_false = false;
 		}
+		// this should never happen, because all arguments which return logical values are operators
 		else
-			result |= atoi(CTEXT(n));
+			result |= (atoi(CTEXT(n)) != 0);
```

Comment:
This should be changed to `result = result || atoi(CTEXT(n)) != 0`. I'm not sure if `|=` works the same way on booleans as it does for ints but it would certainly be cleaner to do it that way.